File: chapter3/java/designpattern/src/main/java/com/designpattern/Pizza.java
package com.designpattern;

import java.util.ArrayList;
import java.util.List;

/**
 * Pizza class
 *
 * @Author: Bridget
 */
public class Pizza {

    private String name;

    private String size;

    private List<String> toppings = new ArrayList<>();

    private List<String> sauces = new ArrayList<>();

    public Pizza(String name) {
        this.name = name;
    }

    public String getSize() {
        return size;
    }

    public void setSize(String size) {
        this.size = size;
    }

    public List<String> getToppings() {
        return toppings;
    }

    public void addToppings(String topping) {
        this.toppings.add(topping);
    }

    public List<String> getSauces() {
        return sauces;
    }

    public void addSauces(String sauce) {
        this.sauces.add(sauce);
    }

    public String getName() {
        return name;
    }

    @Override
    public String toString() {
        return "Pizza{" +
                "name='" + name + '\'' +
                ", size='" + size + '\'' +
                ", toppings=" + toppings +
                ", sauces=" + sauces +
                '}';
    }
}

File: chapter3/typescript/Pizza.ts
/**
 * Pizza TypeScript class
 *
 * @Author: Bridget
 */
 export class Pizza {

    private name: string;
    private size: string = "";
    private toppings: Array<string> = [];
    private sauces: Array<string> = [];

    constructor (name: string){
        this.name = name
    }

    public getSize(): string {
        return this.size;
    }

    public setSize(size: string): void {
        this.size = size;
    }

    public getToppings(): string[] {
        return this.toppings;
    }

    public addToppings(topping: string): void {
        this.toppings.push(topping);
    }

    public getSauces(): string[] {
        return this.sauces;
    }
    public addSauces(sauce: string): void {
        this.sauces.push(sauce);
    }

    public getName(): string {
        return this.name;
    }

    public toString() : string {
        return `Pizza{" +
        "name='" + ${this.name} + '\'' +
        ", size='" + ${this.size}  + '\'' +
        ", toppings=" + ${this.toppings} +
        ", sauces=" + ${this.sauces} +
        '}'`;
    }
}
